aws/table_aws_ec2_instance_availability.go

```diff
@@ -93,17 +97,23 @@ func listAwsAvailableInstanceTypes(ctx context.Context, d *plugin.QueryData, h *
 	}
 
 	input := &ec2.DescribeInstanceTypeOfferingsInput{
-		MaxResults:   aws.Int32(maxLimit),
-		LocationType: types.LocationTypeRegion,
+		MaxResults: aws.Int32(maxLimit),
 	}
 
 	var filters []types.Filter
-	filters = append(filters, types.Filter{Name: aws.String("location"), Values: []string{*region.RegionName}})
+
+	if d.EqualsQualString("location_type") != "" {
+		input.LocationType = types.LocationType(d.EqualsQualString("location_type"))
+	} else {
+		input.LocationType = types.LocationTypeRegion
+		filters = append(filters, types.Filter{Name: aws.String("location"), Values: []string{*region.RegionName}})
```

Comment:
@ParthaI could you please verify if the `filters = append(filters, types.Filter{Name: aws.String("location"), Values: []string{*region.RegionName}})` should be present outside the `if-else` block?

Reply:
@misraved, the code block above is correctly placed based on the [API behavior](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstanceTypeOfferings.html).

- The `location_type` column can have values of `region`, `availability-zone`, `availability-zone-id`, or `outpost`.
- However, with the code changes you suggested, the following queries do not return the expected results:
  - `select * from aws_ec2_instance_availability where location_type = 'availability-zone'`
  - `select * from aws_ec2_instance_availability where location_type = 'availability-zone-id'`
- The issue arises because, while the `location_type` is set to `availability-zone`, the filter (`types.Filter{Name: aws.String("location"), Values: []string{*region.RegionName}}`) is using the region name, leading to empty rows.

Additionally, I noticed that we need `CacheMatch: query_cache.CacheMatchExact` for the `location_type` optional qualifiers. If you run the query `select * from aws_ec2_instance_availability`, the `location_type` will default to `region`. In subsequent queries (e.g., `select * from aws_ec2_instance_availability where location_type = 'availability-zone'`), the cache will be hit, causing a mismatch between the cached result and the query parameter value, which also results in empty rows.

I have pushed a commit in this PR by addressing this behavior.

Thanks!

Reply:
Thanks for the explanation @ParthaI 👍.